pkg/stanza/fileconsumer/file.go

```diff
@@ -109,162 +130,161 @@ func (m *Manager) startPoller(ctx context.Context) {
 	}
-	wg.Wait()
+}
+
+func (m *Manager) handleLostFiles(ctx context.Context) {
+	m.readerLock.Lock()
+	defer m.readerLock.Unlock()
 
-	// Save off any files that were not fully read
 	if m.deleteAfterRead {
-		unfinished := make([]*Reader, 0, len(readers))
-		for _, r := range readers {
+		unfinished := make([]*Reader, 0, len(m.lostReaders))
+		for _, r := range m.lostReaders {
 			if !r.eof {
 				unfinished = append(unfinished, r)
 			}
 		}
-		readers = unfinished
-
-		// If all files were read and deleted then no need to do bookkeeping on readers
-		if len(readers) == 0 {
+		m.lostReaders = unfinished
+		if len(m.lostReaders) == 0 {
 			return
 		}
 	}
+	m.rollReaders(ctx, m.lostReaders)
+	m.lostReaders = make([]*Reader, 0)
+}
 
-	// Any new files that appear should be consumed entirely
-	m.readerFactory.fromBeginning = true
-
+func (m *Manager) rollReaders(ctx context.Context, readers []*Reader) {
+	m.roller.readLostFiles(ctx, readers)
 	m.roller.roll(ctx, readers)
-	m.saveCurrent(readers)
-	m.syncLastPollFiles(ctx)
 }
 
-// makeReaders takes a list of paths, then creates readers from each of those paths,
-// discarding any that have a duplicate fingerprint to other files that have already
-// been read this polling interval
-func (m *Manager) makeReaders(filesPaths []string) []*Reader {
-	// Open the files first to minimize the time between listing and opening
-	files := make([]*os.File, 0, len(filesPaths))
-	for _, path := range filesPaths {
-		if _, ok := m.seenPaths[path]; !ok {
-			if m.readerFactory.fromBeginning {
-				m.Infow("Started watching file", "path", path)
-			} else {
-				m.Infow("Started watching file from end. To read preexisting logs, configure the argument 'start_at' to 'beginning'", "path", path)
-			}
-			m.seenPaths[path] = struct{}{}
-		}
-		file, err := os.Open(path) // #nosec - operator must read in files defined by user
-		if err != nil {
-			m.Debugf("Failed to open file", zap.Error(err))
-			continue
+func (m *Manager) makeReader(filePath string) *Reader {
+	if _, ok := m.seenPaths[filePath]; !ok {
+		if m.readerFactory.fromBeginning {
+			m.Infow("Started watching file", "path", filePath)
+		} else {
+			m.Infow("Startedd watching file from end. To read preexisting logs, configure the argument 'start_at' to 'beginning'", "path", filePath)
 		}
-		files = append(files, file)
+		m.seenPaths[filePath] = struct{}{}
 	}
-
-	// Get fingerprints for each file
-	fps := make([]*Fingerprint, 0, len(files))
-	for _, file := range files {
-		fp, err := m.readerFactory.newFingerprint(file)
-		if err != nil {
-			m.Errorw("Failed creating fingerprint", zap.Error(err))
-			continue
-		}
-		fps = append(fps, fp)
+	file, err := os.Open(filePath) // #nosec - operator must read in files defined by user
+	if err != nil {
+		m.Debugf("Failed to open file", zap.Error(err))
+		return nil
 	}
-
-	// Exclude any empty fingerprints or duplicate fingerprints to avoid doubling up on copy-truncate files
```

Comment:
I think because of my mistake I missed adding this comment inside `m.makeReader()`.
it is addressed here https://github.com/vihas-splunk/opentelemetry-collector-contrib/blob/c72df6bba864e95907cf8a8954b8ff1b8e360f30/pkg/stanza/fileconsumer/file.go#L250-L271

Reply:
I shall update the comments in my next commit